src/main.cpp

```diff
@@ -627,6 +650,25 @@ int main(int argc, char** argv)
     /* Start the event listener */
     drakvuf->loop(timeout);
 
+    if (is_stop_plugin)
+    {
+        PRINT_DEBUG("Beginning stop plugins\n");
+
+        if (drakvuf->stop_plugins(stop_plugin_list) < 0)
+            return drakvuf_exit_code_t::FAIL;
+
+        PRINT_DEBUG("Finished stop plugins\n");
+    }
+
+    if (is_stop_plugin && wait_stop_plugins)
```

Comment:
Why wouldn't the plugins above be stopped already? Doesn't the call to `drakvuf->stop_plugins()` only return once all plugins are safely stopped, ie. there unique_ptr is all set to null, their memory is freed and all the traps they had safely removed with none left pending?

Reply:
`drakvuf->stop_plugins()` just asks every plug-in to stop. Though it doesn't wait them to stop.

E.g. `filedelete` could stop during several seconds. But we could start terminating the loop before `filedelete` stops completely. Or we could wait `filedelete` stops before termination loop.